inc/Engine/Media/Lazyload/CSS/Front/FileResolver.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+namespace WP_Rocket\Engine\Media\Lazyload\CSS\Front;
+
+class FileResolver {
+
+	/**
+	 * Resolves the name from the file from its URL.
+	 *
+	 * @param string $url URL from the file to resolve.
+	 * @return string
+	 */
+	public function resolve( string $url ): string {
+		$parsed_url       = wp_parse_url( $url );
+		$home_parsed_url  = wp_parse_url( site_url(), PHP_URL_HOST );
+		$home_parsed_path = wp_parse_url( site_url(), PHP_URL_PATH );
```

Comment:
Why do 2 calls to wp_parse_url() on the same url instead of one and using the array keys for host and path?